includes/class-amp-theme-support.php

```diff
@@ -98,6 +112,20 @@ class AMP_Theme_Support {
 	 */
 	protected static $support_added_via_option = false;
 
+	/**
+	 * Response cache key.
+	 *
+	 * @var string
+	 */
+	public static $response_cache_key;
```

Comment:
Is this necessary? Aren't references limited to a single method? Since this is a static class it would be better of we limited the state that is persisted in it's class variables IMO.

Reply:
@westonruter I added this public property for the following purposes:

1. to give us the ability to deeply test by knowing what the cache key is and then checking for it in the cache. 
2. to simplify access to the logic within the caching closure, i.e. meaning that we don't have to pass it via `use`.

It serves no other external purpose as part of an API.

Let's think about it.

From a testing point of view, we can _infer_ that we disabled the cache by the state of the post-processor cache.  While that's not a direct test on the response cache, both are within the closure.  Therefore, it's a minimal risk to test by inference.

Hmm, I like that approach better actually.  We get the benefit only exposing what is needed externally while avoided the extra work of resetting persistent state.

I'll make that change.